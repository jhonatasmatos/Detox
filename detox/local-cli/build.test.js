jest.mock('child_process');
jest.mock('../src/utils/logger');
jest.mock('../src/configuration');

describe('build', () => {
  let execSync, composeDetoxConfig, detoxConfig;

  beforeEach(() => {
    detoxConfig = {
      meta: {
        configuration: 'testConfig',
        location: '/etc/detox/config',
      },
      artifactsConfig: {},
      behaviorConfig: {},
      deviceConfig: {},
      sessionConfig: {},
    };

    execSync = require('child_process').execSync;
    composeDetoxConfig = require('../src/configuration').composeDetoxConfig;
    composeDetoxConfig.mockReturnValue(Promise.resolve(detoxConfig));
  });

  it('passes argv to composeConfig', async () => {
    await callCli('./build', 'build -C /etc/.detoxrc.js -c myconf').catch(() => {});

    expect(composeDetoxConfig).toHaveBeenCalledWith({
      argv: expect.objectContaining({
        'config-path': '/etc/.detoxrc.js',
        'configuration': 'myconf',
      }),
    });
  });

  it('runs the build script from the composed device config', async () => {
    detoxConfig.deviceConfig.build = 'yet another command';

    await callCli('./build', 'build');
    expect(execSync).toHaveBeenCalledWith('yet another command', expect.anything());
  });

  it('fails with an error if a build script has not been found', async () => {
    delete detoxConfig.deviceConfig.build;
    await expect(callCli('./build', 'build')).rejects.toThrowErrorMatchingSnapshot();
  });
});
