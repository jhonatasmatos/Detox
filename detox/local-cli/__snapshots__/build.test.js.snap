// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build fails with an error if a build script has not been found 1`] = `"Could not find build script for \\"testConfig\\" in Detox config at: /etc/detox/config"`;
